plugins/llvm/llvm_binary.hpp

```diff
@@ -430,13 +439,14 @@ uint64_t image_entry(const COFFObjectFile* obj) {
 
 template <typename T>
 struct objectfile_image : image {
-    explicit objectfile_image(const T* obj)
-        : arch_(image_arch(obj))
-        , entry_(image_entry(obj))
-        , segments_(seg::read(obj))
-        , symbols_(sym::read(obj))
-        , sections_(sec::read(obj))
-        {}
+    explicit objectfile_image(std::unique_ptr<object::Binary> binary)
+        : arch_(image_arch(dyn_cast<T>(binary.get())))
```

Comment:
This is a trade of that I can't understand: why to call `dyn_cast` five times in the constructor, instead of calling it once at the point where you call the constructor?

Also, calling `get` function on a smart pointer must be avoided if possible. It is possible here.